### DISCUSSION TITLE
Latest update in OBS 3.0 has broken Touch Portal automatic re-connection.

### DISCUSSION THREAD

**Discussion by Rcmanaka** [Palakis/obs-websocket#1185] [Q&A]
After upgrading to OBS 30 the websocket connection to Touch Portal will not reconnect until I either "Kick" the connection OR restart Touch  Portal.

If I have Touch Portal active and then start OBS 30, Touch Portal will not fully communicate with OBS 30 unless I manually restart Touch Portal or alternatively I navigate to the Websocket page in OBS 3.0 and "Kick" the connection.

OBS 29.1.3 re-establishes the connection with Touch Portal correctly.  I can start and stop OBS at will without stopping Touch Portal and the websocket connection is always reconnected.

This behaviour has been confirmed by the Touch Portal developers and they are working on a work-around for their latest version (4x) but I am stuck on their previous version 3.1 build 12 because of Apple's decision to no longer support play store downloads for iPad OS versions less than 12!   My iPad mini (original) has stopped at iOS 9.3.5!

I'd really like the handshake and authorization in OBS 30's Websocket 5 to revert back to the same sequence as in OBS 29.1.3 if at all possible?

Thanks for an otherwise great product!!!

**Comment by tt2468** [comment]
The touch portal developers are encouraged to reach out directly via the Discord server or my email for things like this. If they are seeing an actual regression in functionality, direct contact is the fastest way to get a resolution.
